operator/controllers/module_template_controller.go

```diff
@@ -0,0 +1,78 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/kyma-project/kyma-operator/operator/api/v1alpha1"
+	"github.com/kyma-project/kyma-operator/operator/pkg/remote"
+)
+
+type ModuleTemplateReconciler struct {
+	client.Client
+	record.EventRecorder
+	RequeueIntervals
+}
+
+const CatalogName = "module-catalog"
+
+//nolint:lll
+//+kubebuilder:rbac:groups=operator.kyma-project.io,resources=moduletemplates,verbs=get;list;watch;create;update;patch;onEvent;delete
+//+kubebuilder:rbac:groups=operator.kyma-project.io,resources=moduletemplates/finalizers,verbs=update
+
+func (r *ModuleTemplateReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("Catalog Sync loop starting for", "resource", req.NamespacedName.String())
+
+	// check if kyma resource exists
+	kyma := &v1alpha1.Kyma{}
+	if err := r.Get(ctx, req.NamespacedName, kyma); err != nil {
+		// we'll ignore not-found errors, since they can't be fixed by an immediate
+		// requeue (we'll need to wait for a new notification), and we can get them
+		// on deleted requests.
+		logger.Info(req.NamespacedName.String() + " got deleted!")
```

Comment:
This log message is a bit misleading, or? I think only if it's not found error, we can safely say it's got deleted.

Reply:
You are right, I missed this. Will add an exception case to the err here as I forgot